sys/linux/vnet.txt

```diff
@@ -733,9 +738,49 @@ tcp_option [
 	fastopen	tcp_fastopen_option
 	exp_fastopen	tcp_exp_fastopen_option
 	exp_smc		tcp_exp_smc_option
+    mptcp       tcp_mptcp_option
 ] [varlen]
 
-tcp_option_types = TCPOPT_NOP, TCPOPT_EOL, TCPOPT_MSS, TCPOPT_WINDOW, TCPOPT_SACK_PERM, TCPOPT_SACK, TCPOPT_TIMESTAMP, TCPOPT_MD5SIG, TCPOPT_FASTOPEN, TCPOPT_EXP
+mptcp_option [
+    generic     mptcp_generic_option
+#    syn         mptcp_syn_option
+#    synack      mptcp_synack_option
+#    ack         mptcp_ack_option
+    capable     mptcp_capable_option
+#    data_ack    mptcp_data_ack_option
+#    add_addr    mptcp_add_addr_option
+#    mp_join     mptcp_mp_join_option
+#    mp_fail     mptcp_mp_fail_option
+#    mp_flcose   mptcp_mp_fclose_option
+#    remove_addr mptcp_remove_addr_option
+#    mp_prio     mptcp_mp_prio_option
+] [varlen]
+
+tcp_option_types = TCPOPT_NOP, TCPOPT_EOL, TCPOPT_MSS, TCPOPT_WINDOW, TCPOPT_SACK_PERM, TCPOPT_SACK, TCPOPT_TIMESTAMP, TCPOPT_MD5SIG, TCPOPT_FASTOPEN, TCPOPT_EXP, TCPOPT_MPTCP
+
+mptcp_option_types = OPTION_TYPE_SYN, OPTION_TYPE_SYNACK, OPTION_TYPE_ACK, OPTION_MP_CAPABLE, OPTION_DATA_ACK, OPTION_ADD_ADDR, OPTION_MP_JOIN, OPTION_MP_FAIL, OPTION_MP_FCLOSE, OPTION_REMOVE_ADDR, OPTION_MP_PRIO
+
+# https://tools.ietf.org/html/rfc6824#section-3
+tcp_mptcp_option {
```

Comment:
I think this should reuse `mptcp_option`.

Reply:
What do you mean exactly?

Reply:
I don't understand why you need this struct and don't just use `mptcp_option` instead in `tcp_option`.

Reply:
Maybe. I don't know the specifics of this API, but `tcp_mptcp_option` looks very similar to `mptcp_generic_option` and that's confusing. If the `tcp_mptcp_option` struct is the same thing as the `tcp_option` union, then merge them.

Reply:
`mptcp_generic_option` looks similar to `tcp_generic_option`because i tried to adapt the structure from all tcp structs and unions to mptcp. 
`tcp_mptcp_option` looks similiar to `tcp_mss_option` as MPTCP is also a TCP option but with some sub types like MP_CAPABLE or ADD_ADDR

Reply:
OK, could you address the other comment about removing `mptcp_header` and `mptcp_packet` first and I'll take another look at this pull request then.

Reply:
I already answered this question here: 
https://github.com/google/syzkaller/pull/848#issuecomment-443729375
MPTCP is a TCP Option. Option Kind 30 with several sub types. According to this, we don't need `mptcp_header` and `mptcp_packet`. It's just a `tcp_packet` with a `tcp_header`